File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.length();
        int len = 0;
        string ans = "";
        for (int i = 0; i < n; i++) {

            // odd length
            int left = i;
            int right = i;

            while (left >= 0 && right < n && s[left] == s[right]) {
                left--;
                right++;
            }
            if ((right - left - 1) > len) {
                len = right - left - 1;
                ans = s.substr(left + 1, right - left - 1);
            }

            // even length
            left = i;
            right = i + 1;
            while (left >= 0 && right < n && s[left] == s[right]) {
                left--;
                right++;
            }
            if ((right - left - 1) > len) {
                len = right - left - 1;
                ans = s.substr(left + 1, right - left - 1);
            }
        }
        return ans;
    }
};

File: Difficulty: Medium/Longest Palindromic Substring/longest-palindromic-substring.py
class Solution:
    def getLongestPal(self, s):
        # code here
        ans = ""
        l = 0
        for i in range(len(s)):
            # odd
            left = i
            right = i
            while ((left >= 0 and right < len(s)) and (s[left] == s[right])):
                left -= 1
                right += 1
            if right - left - 1 > l:
                l = right - left - 1
                ans = s[left+1 : right]
    
            # even
            left = i
            right = i + 1
            while (left >= 0 and right < len(s)) and (s[left] == s[right]):
                left -= 1
                right += 1
            if right - left - 1 > l:
                l = right - left - 1
                ans = s[left+1 : right]
        return ans